staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go

```diff
@@ -121,37 +188,198 @@ func (c *client) call(method string, url string, body io.Reader) ([]byte, error)
 }
 
 func (c *client) Instances() ([]*instanceResp, error) {
-	url := c.chooseURL()
-	Scope.Debug("nacos url:" + url)
+	var fetcher func() (map[string][]*instance, error)
+	if c.fetchAllNamespaces {
+		fetcher = c.allNamespacesInstances
+	} else {
+		fetcher = func() (map[string][]*instance, error) {
+			return c.namespacedGroupedInstances(c.namespaceId, c.group)
+		}
+	}
+	m, err := fetcher()
+	if err != nil {
+		Scope.Errorf("do get instances failed: %s", err)
+		return nil, err
+	}
+	resp := make([]*instanceResp, 0, len(m))
+	for svc, instances := range m {
+		resp = append(resp, &instanceResp{
+			Dom:   svc,
+			Hosts: instances,
+		})
+	}
+	return resp, nil
+}
+
+func (c *client) pagingListServices(namespaceId, groupName string, pageNo int) (*serviceResp, error) {
+	var sr serviceResp
+	param := map[string]string{
+		"namespaceId": namespaceId,
+		"groupName":   groupName,
+		"pageSize":    fmt.Sprintf("%d", serviceListPageSize),
+		"pageNo":      fmt.Sprintf("%d", pageNo),
+	}
+	c.injectAuthParam(param)
+	resp, err := c.call(serviceListAPI, http.MethodGet, c.headers, param, nil)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(resp, &sr); err != nil {
+		return nil, err
+	}
+	return &sr, nil
+}
+
+func (c *client) listServices(namespaceId, groupName string) ([]string, error) {
+	probeServiceResp, err := c.pagingListServices(namespaceId, groupName, 1)
+	if err != nil {
+		return nil, err
+	}
+	doms := probeServiceResp.Doms
+	if probeServiceResp.Count > serviceListPageSize {
+		pageCount := probeServiceResp.Count/serviceListPageSize + 1
```

Comment:
I have some doubts that if `Count == 2000` and we get `pageCount == 3` and then whether we'll get a err when `page == 3`?